app/extensions/safe/server-routes/safe.js

```diff
@@ -28,34 +31,53 @@ const safeRoute = ( store ) => ( {
 
             if ( headers.range )
             {
+	        isRangeReq = true;
                 const range = headers.range;
-                const rangeArray =
+                rangeArray =
                   range.substring( BYTES.length, range.length )
-                      .replace( /['"]+/g, '' )
-                      .split( '-' );
-
-                start = rangeArray[0] ? parseInt( rangeArray[0], 10 ) : null;
-                end = rangeArray[1] ? parseInt( rangeArray[1], 10 ) : null;
-
-                // we need to separate out 0- length requests which are not partial content
-                if ( start && start !== 0 )
-                {
-                    // should have a start !== 0
-                    isRangeReq = true;
-                }
-
-                if ( start === 0 && end )
-                {
-                    isRangeReq = true;
-                }
+                      .split( ',' )
```

Comment:
@hunterlester it's looking like this function is getting loooong.

Can we start splitting this out (maybe into another file if it makes sense?) To keep the overall server handling clean.

We should then be able to more easily test that it's working correctly too 👍